archive/zip.ksy

```diff
@@ -36,22 +36,34 @@ doc-ref:
 seq:
   - id: sections
     type: pk_section
-    repeat: eos
+    repeat: until
+    repeat-until: _.section_type == section_types::end_of_central_dir
 types:
   pk_section:
     seq:
       - id: magic
         contents: 'PK'
       - id: section_type
         type: u2
+        enum: section_types
       - id: body
         type:
           switch-on: section_type
           cases:
-            0x0201: central_dir_entry
-            0x0403: local_file
-            0x0605: end_of_central_dir
-            0x0807: data_descriptor
+            section_types::central_dir_entry: central_dir_entry
+            section_types::local_file: local_file
+            section_types::end_of_central_dir: end_of_central_dir
+            section_types::data_descriptor: data_descriptor
+            section_types::archive_extra_data: archive_extra_data
+            section_types::digital_signature: digital_signature
+            section_types::zip64_end_of_central_dir: zip64_end_of_central_directory
+            section_types::zip64_end_of_central_dir_locator: zip64_end_of_central_directory_locator
```

Comment:
I would use `central_dir` everywhere to make the spec consistent, rather than interspersing it with expanded ~~`central_directory`~~.

```diff
-            section_types::zip64_end_of_central_dir: zip64_end_of_central_directory
-            section_types::zip64_end_of_central_dir_locator: zip64_end_of_central_directory_locator
+            section_types::zip64_end_of_central_dir: zip64_end_of_central_dir
+            section_types::zip64_end_of_central_dir_locator: zip64_end_of_central_dir_locator
```

Please rename all relevant fields and types (and their usages) accordingly.